test/utils.test.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable sonarjs/no-duplicate-string */
+const path = require('path');
+const Generator = require('../lib/generator');
+const log = require('loglevel');
+const utils = jest.requireActual('../lib/utils');
+
+describe('Utils', () => {
+  describe('#getTemplateDetails', () => {
+    let resolvePkg, resolveFrom;
+    const templateNpmName = 'nameOfTestTemplate';
+
+    beforeEach(() => {
+      resolvePkg = require('resolve-pkg');
+      resolveFrom = require('resolve-from');
+      jest.mock(path.resolve('./testTemplate', 'package.json'), () => ({ name: 'nameOfTestTemplate' }), { virtual: true });
+      jest.mock(path.resolve(Generator.DEFAULT_TEMPLATES_DIR, templateNpmName, 'package.json'), () => ({ name: 'nameOfTestTemplate' }), { virtual: true });
+    });
+
+    it('works with a file system path', () => {
+      log.debug = jest.fn();
+      utils.isFileSystemPath = jest.fn(() => true);
+      const templatePath = './testTemplate';
+      const result = utils.getTemplateDetails(templatePath, 'package.json');
+      expect(log.debug).toHaveBeenCalledWith('Remember that your local template must have its own node_modules installed first, \"npm install\" is not triggered by the generator.');
+      expect(result).toStrictEqual({
+        name: templateNpmName, 
+        pkgPath: path.resolve('./testTemplate')
+      });
+    });
+
+    it('works with an npm package', () => {
+      log.debug = jest.fn();
+      utils.isFileSystemPath = jest.fn(() => false);
+      const packagePath = path.join(Generator.DEFAULT_TEMPLATES_DIR, templateNpmName);
+      resolvePkg.__resolvePkgValue = packagePath;
+      const result = utils.getTemplateDetails(templateNpmName, 'package.json');
+      expect(log.debug).not.toHaveBeenCalledWith('Remember that your local template must have its own node_modules installed first, \"npm install\" is not triggered by the generator.');
```

Comment:
Not for this PR, but we should move all texts hardcoded in code to separate file. I mean, by changing the text in code you have to change the tests immediately, and this a problem for me.

Reply:
Love the idea https://github.com/asyncapi/generator/issues/533